vm/src/hint_processor/builtin_hint_processor/segments.rs

```
@@ -19,11 +19,35 @@ pub fn relocate_segment(
     ap_tracking: &ApTracking,
 ) -> Result<(), HintError> {
     let src_ptr = get_ptr_from_var_name("src_ptr", vm, ids_data, ap_tracking)?;
-    #[cfg(not(feature = "extensive_hints"))]
-    let dest_ptr = get_ptr_from_var_name("dest_ptr", vm, ids_data, ap_tracking)?;
-    #[cfg(feature = "extensive_hints")]
-    let dest_ptr = crate::hint_processor::builtin_hint_processor::hint_utils::get_maybe_relocatable_from_var_name("dest_ptr", vm, ids_data, ap_tracking)?;
-
+    // Bugfix: this is a workaround for an issue in the vm related to the way it computes
+    // `dest_ptr` in `segment_arena.cairo`. For some reason it sets `dest_ptr` to 0 (felt) instead
+    // of a valid relocatable value.
+    // As this hint is used in other places, we need to determine if we are in `segments_arena`
+    // first. We check this by determining if the "infos" variable is declared.
+    // If it is the case, we simply recompute `dest_ptr` manually.
+    let dest_ptr = if let Ok(infos) = get_ptr_from_var_name("infos", vm, ids_data, ap_tracking) {
```

Maybe this can be improved with something like:
```rs
   fn is_segment_arena(ids_data: &HashMap<String, HintReference>) -> bool {
        return ids_data.contains_key("infos")
            && ids_data
                .get("infos")
                .expect("Must contain infos")
                .cairo_type
                .expect("Must have a type")
                .ends_with("SegmentInfo*");
    }
```

---

I would avoid trying to match strings when checking for types. There could be future updates which changes `SegmentInfo*` with a different name. This would end up not only in a silent error, but a very difficult one to spot.

---

Noted!